compiler/src/iree/compiler/InputConversion/MHLO/ConvertCollectiveOps.cpp

```diff
@@ -275,15 +313,39 @@ static DenseIntElementsAttr convertToRankGroupsByCrossReplicaAndPartition(
   return DenseIntElementsAttr::get(type, newValues);
 }
 
+enum class CollectiveOpGroupMode {
+  CrossReplica,
+  CrossPartition,
+  CrossReplicaAndPartition,
+  FlattenedIds,
+};
+
+static CollectiveOpGroupMode getCollectiveOpGroupMode(
```

Comment:
Since we are putting all the modes here, it would be nice to have a table as documentation that shows the support modes for each collective op.